### PR TITLE
Update for L2 cache driver and pre-fetcher.

### PR DESCRIPTION
**Pull request by NandkumarJoshi**
Add new register defines. Removes references to cache driver 'handle' and related APIs, since there will be only a single L2 cache device per core design.
Following are the generated defines in bsp header, referenced in L2 cache driver.

#define METAL_SIFIVE_CCACHE0_INTERRUPTID0 31

#define METAL_SIFIVE_CCACHE0_INTERRUPTID1 32

#define METAL_SIFIVE_CCACHE0_INTERRUPTID2 33

#define METAL_SIFIVE_CCACHE0_INTERRUPTID3 34

#define METAL_SIFIVE_CCACHE0_INTERRUPT_PARENT &__metal_dt_interrupt_controller_c000000.controller

#define METAL_CACHE_DRIVER_PREFIX sifive_ccache0

#define METAL_SIFIVE_CCACHE0_PERFMON_COUNTERS 6

### CODE REVIEW

File: bare_header/sifive_l2hwpf0.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  void emit_offsets() {
+    if (dtb.match(std::regex(compat_string), [](const node n) {}) != 0) {
+      emit_compat();
+      emit_offset("REG0", 0x00);
+      emit_offset("REG1", 0x04);
+
+      os << std::endl;
+    }
+  }
```
Comment (bsousi5) [inline bare_header/sifive_l2hwpf0.h:37]: We also need these definitions for these values for init
        sifive,spf-queue-entries = <16>;
        sifive,spf-window-bits = <6>;
        sifive,spf-distance-bits = <6>;
        sifive,spf-streams = <8>;

File: metal_header/sifive_ccache0.c++ [hunk 2]
```diff
@@ -13,76 +13,42 @@ void sifive_ccache0::include_headers() {
...
-    extern_inlines.pop_front();
-    emit_inline_def(func, "sifive_ccache0");
-    delete func;
-  }
-  os << "\n";
-}
+    emit_def("METAL_SIFIVE_CCACHE0_INTERRUPTID0", std::to_string(i0));
+    emit_def("METAL_SIFIVE_CCACHE0_INTERRUPTID1", std::to_string(i1));
+    emit_def("METAL_SIFIVE_CCACHE0_INTERRUPTID2", std::to_string(i2));
+    emit_def("METAL_SIFIVE_CCACHE0_INTERRUPTID3", std::to_string(i3));
```
Comment (bsousi5) [inline metal_header/sifive_ccache0.c++:31]: The number of interrupt are not necessary 4 all the time, and is based on the number of l2 banks.
You will need to count the number of interrupts and create the appropriate number of IDs

File: metal_header/sifive_ccache0.c++ [hunk 3]
```diff
@@ -13,76 +13,42 @@ void sifive_ccache0::include_headers() {
...
 
   dtb.match(std::regex(compat_string), [&](node n) {
-    if (count == 0) {
-      func = create_inline_dec("control_base", "uintptr_t",
-                               "struct metal_cache *cache");
-      extern_inlines.push_back(func);
+    uint32_t i0 = 0, i1 = 0, i2 = 0, i3 = 0;
+    if (n.field_exists("interrupts")) {
+      /* Get interrupt source IDs */
+      i0 = n.get_fields<uint32_t>("interrupts")[0];
```
Comment (bsousi5) [inline metal_header/sifive_ccache0.c++:22]: The number of interrupt are not necessary 4 all the time, and is based on the number of l2 banks.
You will need to count the number of interrupts and get the appropriate IDs

**Comment by NandkumarJoshi** [comment]
Closing this request. Raised new one against v20.05.01-branch.
https://github.com/sifive/freedom-devicetree-tools/pull/232

**Pull request closed without merging**